MAVProxy/modules/mavproxy_chat/assistant_setup/get_mode_mapping.json

```diff
@@ -0,0 +1,15 @@
+{
+  "type": "function",
+  "function": {
+    "name": "get_mode_mapping",

```

Comment:
I can't help but wonder if it would be nicer to generate all of this from comments in the Python code rather than explicit documentation like this.

Advantages to human programmers in that.  Especially when your editor is smart enough to use those comments to prompt in the user interface.  Additional advantage that this documentation won't get out-of-date with the code, then.

Still, carry on :-)

Reply:
OK, that would be cool.  The json files are slightly difficult to get right the first time you make them but after that it gets easier.  What's odd about them is that you only need to include the list of arguments that the assistant must fill in.  The return values can be whatever you want.. somehow the assistant just deals with whatever it gets.